### ISSUE TITLE
Suggestion: "New MFS" option?

### ISSUE THREAD

**Issue by bleonard252** [ipfs-shipyard/ipfs-webui#855]
Could we add a New MFS option to the Files browser? It would:

* Create a directory in IPFS
* Create a file record in the directory to tell the webui to use the folder as an MFS and update the record if edits are made (and prevent edits if the IPNS key is missing!)
* Create an IPNS record to the IPFS directory
* Update the IPNS record anytime a change is made to the file

I'll go ahead and find where to start, so help me out in bleonard252/ipfs-webui please!

**Comment by hacdias** [comment]
@bleonard252 hello! I'm not sure if I understand what you want: the current Web UI (2.0) already uses MFS in the files section:

![image](https://user-images.githubusercontent.com/5447088/46836417-43797300-cda9-11e8-8b1d-b27289462207.png)

If you click 'Add to IPFS', you'll have an option to create a new directory/folder. 😄 Hope it helps!

**Comment by bleonard252** [comment]
Yes, the auto-updating IPNS is the part I'm looking for. Keep me posted, please!

**Comment by lidel** [comment]
This would be a cool feature to have, but requires UI design first.
FYSA it is already tracked in https://github.com/ipfs/ipfs-gui/issues/52
A good start would be to research and propose solution to open questions listed there.

**Comment by lidel** [comment]
FYSA manual publishing to IPNS is being added in https://github.com/ipfs/ipfs-webui/pull/1857
